util/subtree_linux.go

```diff
@@ -58,5 +67,33 @@ func cleanup(key string, logger grip.Journaler) error {
 			}
 		}
 	}
+
+	// Retry listing processes until all have successfully exited
+	ctx := context.Background()
+	err = Retry(
+		ctx,
+		func() (bool, error) {
+			pids, err = listProc()
+			if err != nil {
+				return false, err
+			}
+			if len(pids) == 0 {
+				return false, nil
+			}
+			return true, nil
+		},
+		processCleanupAttempts,
+		processCleanupTimeoutMin,
+		processCleanupTimeoutMax,
+	)
+	if err != nil {
+		return err
+	}
+
+	// Log each process that was not cleaned up
+	for pid := range pids {
+		logger.Infof("Failed to clean up process &d", pid)
```

Comment:
Should we truncate this if the number of pids is very long? It also might make sense to build up a single string first and then log that string rather than calling the logger many times.

Reply:
Summing up our chat -- killing processes already logs each one separately, and we don't want to truncate the list of zombied pids even if it gets long (might be misleading if a pid is never logged). We'll keep it this way for now!